packages/schematics/angular/utility/ast-utils.ts

```diff
@@ -574,6 +574,59 @@ export function isImported(source: ts.SourceFile,
   return matchingNodes.length > 0;
 }
 
+function getEnvironmentSpecifierName(namedBindings: ts.NamedImports): string | null {
+  for (const specifier of namedBindings.elements) {
+    // `propertyName` is defined if the specifier
+    // has an aliased import.
+    const name = specifier.propertyName || specifier.name;
+
+    // Check the default export name.
+    if (name.text.includes('environment')) {
+      return specifier.name.text;
+    }
+  }
+
+  return null;
+}
+
+/**
+ * This function returns the name of the environment export
+ * whether this export is aliased or not. If the environment file
+ * is not imported, then it will return `null`.
+ */
+export function getEnvironmentExportName(source: ts.SourceFile): string | null {
+  const allNodes = getSourceNodes(source);
+
+  const importDeclaration = allNodes
+    .filter(node => node.kind === ts.SyntaxKind.ImportDeclaration)
+    .filter(
+      (declaration: ts.ImportDeclaration) =>
+        declaration.moduleSpecifier.kind === ts.SyntaxKind.StringLiteral,
+    )
+    .find((declaration: ts.ImportDeclaration) => {
+      if (!declaration.importClause) {
+        return false;
+      }
+
+      // Find specifier that contains `environment` keyword in its text.
+      // Whether it's `environment` or `environment as env`.
+      const nodes = findNodes(declaration, ts.SyntaxKind.ImportSpecifier).filter(node =>
+        node.getText().includes('environment'),
```

Comment:
`getText` can be an expensive operation.
Can you move the named bindings analysis up to this point and use it to check the values?

Reply:
I hope I understand you correctly. :slightly_smiling_face: